userland/libtock/adc.c

```diff
@@ -1,49 +1,279 @@
+// ADC interface
+
 #include <stdint.h>
+#include <stdio.h>
 
 #include "tock.h"
 #include "adc.h"
 
-struct adc_data {
-  int reading;
+// used for creating synchronous versions of functions
+//
+// fired - set when the callback has been called
+// channel - channel that the collected sample corresponds to
+// sample - collected sample value, valid if single sample operation
+// length - number of collected sample values, valid if multiple sample
+//          operation
+// buffer - pointer to buffer filled with samples, valid if multiple sample
+//          operation
+// error - set to FAIL if an invalid callback type is detected
+typedef struct {
   bool fired;
-};
+  uint8_t channel;
+  uint16_t sample;
+  uint32_t length;
+  uint16_t* buffer;
+  int error;
+} adc_data_t;
```

Comment:
I'm confused. The comment and the struct elements are already in the same order.